src/DummyQueryContext.cs

```diff
@@ -1,24 +1,80 @@
-﻿using System.Linq.Expressions;
+﻿using System;
+using System.Linq.Expressions;
 using Starcounter.Linq.Visitors;
 
 namespace Starcounter.Linq
 {
     public interface IQueryContext
     {
         object Execute(Expression expression);
+        object Execute<TResult>(Expression expression);
         string GetQuery(Expression expression);
     }
 
     public class QueryContext<T> : IQueryContext
     {
-        public object Execute(Expression expression)
+        public object Execute(Expression expression) => Execute<T>(expression);
+
+        public object Execute<TResult>(Expression expression)
         {
             var query = new QueryBuilder<T>();
 
             RootVisitor<T>.Instance.Visit(expression, query);
             var sql = query.BuildSqlString();
             var variables = query.GetVariables();
-            return Db.SlowSQL<T>(sql, variables); ;
+
+            if (typeof(TResult).IsGenericType)
+            {
+                return Db.SQL<T>(sql, variables);
```

Comment:
.SQL cannot handle literals, Skip and Take etc will fail.
Merge the two if-statements IMO

Reply:
Yeah.. somehow I forgot about it.

Reply:
Which approach would be better - to use `Db.SlowSQL` or to change Take and Skip implementation to use variables instead of literals?